faucet.go

```diff
@@ -0,0 +1,674 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+// faucet is an Ether faucet backed by a light client.
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/ethdb/leveldb"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	startKey = "StartBlock"
+)
+
+var (
+	ethRpcFlag = flag.String("ethrpc", "", "Ethereum mainnet rpc URL for ethclient to get address balance")
+	sepRpcFlag = flag.String("sepwsrpc", "", "Ethereum Sepolia websocket URL for ethclient to get swap transaction")
+	wsRpcFlag  = flag.String("wsrpc", "", "SWC websocket URL for ethclient to submit faucet or swap transactions")
```

Comment:
The naming of sepRpcFlag and wsRpcFlag is confusing because sepRpcFlag actually represents a WebSocket RPC but doesn’t include `ws` in the name, while wsRpcFlag is specifically for Sepolia..

Reply:
it’s also necessary to update wsRpcFlag for consistency